src/microscope_tiling_basler_camera/src/microscope_tiling_basler_camera/main.py

```diff
@@ -0,0 +1,152 @@
+"""Example code entry point."""
+
+import os
+import numpy as np
+import cv2
+
+from numpy.typing import NDArray
+from cv2.typing import MatLike
+from zaber_motion.ascii import Connection, AxisGroup
+from zaber_motion import Measurement, Units
+from zaber_motion.microscopy import Microscope
+from basler_camera_wrapper import (
+    BaslerCameraWrapper,
+    ImageCaptureError,
+)
+from example_util import is_even, try_stitch_images, join_tiles
+from path_builder import PathBuilder
+
+# user-specified params
+SERIAL_PORT: str = "/dev/tty.usbserial-AB0PG6A5"
+PIXEL_WIDTH_MICRONS: float = 3.0177054668168553
+PIXEL_HEIGHT_MICRONS: float = 3.0177054668168553
+CAMERA_ROTATION_RAD: float = 0.045226578
+
+# TODO: reset point values to 0
+TOP_LEFT: NDArray[np.float64] = np.array([51.6696875, 48.70765625, 19.687559])
+BOTTOM_RIGHT: NDArray[np.float64] = np.array([55.06625, 44.0253125, 19.687559])
+POINTS_UNITS: Units = Units.LENGTH_MILLIMETRES  # units for TOP_LEFT and BOTTOM_RIGHT
+OVERLAP_H: float = 0.5
+OVERLAP_V: float = 0.5
+
+# image capture
+SAVE_FOLDER: str = "tiles"
+RUN_BEST_EFFORT_STITCHING: bool = True
+RUN_NAIVE_TILING: bool = True
+
+# non user-controlled params
+CENTRE: NDArray[np.float64] = TOP_LEFT + (BOTTOM_RIGHT - TOP_LEFT) / 2.0
+EPSILON: float = 0.0000001
+
+
+def main() -> None:
+    """example entry point"""
+    check_user_specified_params()
+
+    if not os.path.exists(SAVE_FOLDER):
+        os.makedirs(SAVE_FOLDER)
+
+    with Connection.open_serial_port(SERIAL_PORT) as connection:
+        connection.detect_devices(identify_devices=True)
+        microscope: Microscope = Microscope.find(connection)
+        if microscope.plate is not None:
+            plate: AxisGroup = microscope.plate
+        else:
+            raise RuntimeError("Microscope plate is None")
+
+        print("Homing microscope plate")
+        plate.home()
+
+        camera: BaslerCameraWrapper = BaslerCameraWrapper()
+
+        path_builder: PathBuilder = PathBuilder(
+            PIXEL_WIDTH_MICRONS,
+            PIXEL_HEIGHT_MICRONS,
+            CAMERA_ROTATION_RAD,
+            camera.get_frame_width(),
+            camera.get_frame_height(),
+        )
+        tiling_path = path_builder.get_path_snake(
+            TOP_LEFT, BOTTOM_RIGHT, POINTS_UNITS, OVERLAP_H, OVERLAP_V
+        )
+
+        tiles = capture_images(tiling_path, camera, plate)
+        num_rows: int = len(tiling_path)
+
+        if RUN_BEST_EFFORT_STITCHING:
+            try:
+                try_stitch_images(tiles, num_rows)
+            except RuntimeError:
+                print("Stitching failed")
```

Comment:
probably worth printing the exception or at least taking over the message, otherwise we don't know what's wrong.

Reply:
fixed